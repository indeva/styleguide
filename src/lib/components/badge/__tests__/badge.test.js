import React from 'react';
import renderer from 'react-test-renderer';
import Badge from '..';

describe('Badge', () => {
  it('should render default badge as neutral', () => {
    const component = renderer.create(<Badge>test</Badge>);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should render neutral badge', () => {
    const component = renderer.create(<Badge type="neutral">test</Badge>);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should render brand badge', () => {
    const component = renderer.create(<Badge type="brand">test</Badge>);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should render success badge', () => {
    const component = renderer.create(<Badge type="success">test</Badge>);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should render warning badge', () => {
    const component = renderer.create(<Badge type="warning">test</Badge>);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should render danger badge', () => {
    const component = renderer.create(<Badge type="danger">test</Badge>);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should render with className', () => {
    const component = renderer.create(<Badge className="test">test</Badge>);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
