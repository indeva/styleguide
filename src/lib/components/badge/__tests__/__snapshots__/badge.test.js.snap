// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge should render brand badge 1`] = `
<span
  className="badge badge--brand"
>
  test
</span>
`;

exports[`Badge should render danger badge 1`] = `
<span
  className="badge badge--danger"
>
  test
</span>
`;

exports[`Badge should render default badge as neutral 1`] = `
<span
  className="badge badge--neutral"
>
  test
</span>
`;

exports[`Badge should render neutral badge 1`] = `
<span
  className="badge badge--neutral"
>
  test
</span>
`;

exports[`Badge should render success badge 1`] = `
<span
  className="badge badge--success"
>
  test
</span>
`;

exports[`Badge should render warning badge 1`] = `
<span
  className="badge badge--warning"
>
  test
</span>
`;

exports[`Badge should render with className 1`] = `
<span
  className="badge badge--neutral test"
>
  test
</span>
`;
